ngraph/core/reference/include/ngraph/runtime/reference/interpolate.hpp

```diff
@@ -454,7 +498,88 @@ namespace ngraph
             }
 
             template <typename T>
-            void InterpolateEval<T>::linear_onnx_func(const T* input_data, T* out)
+            void InterpolateEval<T>::linear_onnx5D_func(const T* input_data, T* out)
```

Comment:
I propose to write the common logic for any Nd tensors. Looks like the 90% of this function duplicate the funciton for 4D tensors.

Reply:
1) It is an implementation of 5D case of `linear_onnx` mode.
2) The `linear_onnx` mode is a linear interpolation in the ONNX sense.
3) ONNX runtime doesn't support ranks greater than 5.
4) The common logic for Nd tensors is not obvious. 
5) We cannot formulate such common logic for Nd tensors, because such common logic is unknown for ONNX writers.
6) A possible future ONNX implementation of greater ranks case may differ from the common logic we have now formulated.